descwl_coadd/tests/test_coadd.py

```diff
@@ -163,7 +170,71 @@ def test_coadds_with_interpolator_smoke(buff):
         assert coadd_data['coadd_psf_exp'].image.array.shape == psf_dims
 
         assert np.all(np.isfinite(coadd_data['coadd_psf_exp'].image.array))
-        assert np.all(coadd_data['coadd_mfrac_exp'].image.array == 0)
+
+        if not bad_columns:
+            assert np.all(coadd_data['coadd_mfrac_exp'].image.array == 0)
+
+
+@pytest.mark.parametrize('buff', [4, 5])
+def test_coadd_equality_with_dm_interpolator(buff):
+    """Test that the DM-implementation of the interpolator behaves identical.
+    """
+    rng = np.random.RandomState(123)
+
+    coadd_dim = 101
+    psf_dim = 51
+
+    bands = ['r', 'i', 'z']
+    sim_data = _make_sim(
+        rng=rng, psf_type='gauss', bands=bands,
+        coadd_dim=coadd_dim, psf_dim=psf_dim,
+        stars=False, bad_columns=True,
+    )
+
+    our_interpolator = CTInterpolator(buff=buff)
+
+    config = CloughTocher2DInterpolateTask.ConfigClass()
+    config.interpLength = buff
+    config.badMaskPlanes = our_interpolator.bad_mask_planes
+
+    dm_interpolator = CloughTocher2DInterpolateTask(config=config)
+
+    # coadd each band separately
+    bdata = sim_data['band_data']
+    for band in bands:
+        assert band in bdata
+        exps = bdata[band]
+
+        # Check that there are pixels to be interpolated.
+        bit_mask = exps[0].mask.getPlaneBitMask(our_interpolator.bad_mask_planes)
+        assert any((exp.mask.array & bit_mask).any() for exp in exps)
+
+        # Since interpolation changes the exposures in place, make a deep
+        # copy before making the coadds.
+        coadd_data = make_coadd(
+            exps=[exp.clone() for exp in exps],
+            coadd_wcs=sim_data['coadd_wcs'],
+            coadd_bbox=sim_data['coadd_bbox'],
+            psf_dims=sim_data['psf_dims'],
+            rng=rng,
+            remove_poisson=False,  # no object poisson noise in sims
+            interpolator=our_interpolator,
+        )
+
+        dm_coadd_data = make_coadd(
+            exps=exps,
+            coadd_wcs=sim_data['coadd_wcs'],
+            coadd_bbox=sim_data['coadd_bbox'],
+            psf_dims=sim_data['psf_dims'],
+            rng=rng,
+            remove_poisson=False,  # no object poisson noise in sims
+            interpolator=dm_interpolator,
+        )
+
+        # Compare only the main exposure since the noise realizations are
+        # different and mfrac is not affected by the interpolator.
+        assert (coadd_data['coadd_exp'].image.array ==
+                dm_coadd_data['coadd_exp'].image.array).all()
```

Comment:
```suggestion
        np.testing.assert_array_equal(
            coadd_data['coadd_exp'].image.array,
            dm_coadd_data['coadd_exp'].image.array,
        )
```

This will gives us a nicer diff so we can get a better idea of what the issue is.